lib/modules/dosomething/dosomething_fact/dosomething_fact.module

```diff
@@ -185,7 +185,7 @@ function dosomething_fact_get_formatted_sources($sources) {
     return NULL;
   }
 
-  if (!is_array($sources[0])) {
+  if (array_key_exists("raw", $sources)) {
```

Comment:
just explained to diego but ill try to do it here

The problem is the arrays we're checking for could have a 0, 1, 2 etc normal array structure, or JUST be formatted and raw. 

For arrays with formatted and raw, searching for [0] tosses an error. This performs the same check without an error.